### PR TITLE
Avoid printing the progress bar if the destination is not a TTY.

### LINKED ISSUE #60: error when redirecting output
Under Linux, with node 6.3.0. Also happens on master:

`autocannon -d 1 example.com > out 2>&1`

Result - error in node_modules/progress/lib/node-progress.js:177: 

`TypeError: this.stream.clearLine is not a function`

Happens only if both stdout and stderr are redirected.

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #60.

@davidmarkclements @thekemkid please review.

### CODE REVIEW

File: lib/progressTracker.js [hunk 1]
```diff
@@ -26,54 +26,60 @@ function track (instance, opts) {
...
-      instance.on('tick', () => {
-        progressBar.tick()
-      })
-
-      process.once('SIGINT', () => {
-        progressBar.tick(iOpts.duration - 1)
-      })
+      // if switch needed needed to avoid
+      // https://github.com/mcollina/autocannon/issues/60
+      if (opts.outputStream.isTTY) {
```
Comment (thekemkid) [inline lib/progressTracker.js]: We could turn the progress bar output into its own function, and in that case, we could return early if the output stream isn't a tty, like below

```js
if (!opts.outputStream.isTTY) return
```

Not a worry if you don't want to, only a minor thing I realised

File: test/cli.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+const lines = [
+  /Running 1s test @ .*$/,
+  /10 connections.*$/,
+  /$/,
+  /Stat.*Avg.*Stdev.*Max.*$/,
+  /Latency \(ms\).*$/,
+  /Req\/Sec.*$/,
+  /Bytes\/Sec.*$/,
+  /$/,
+  /\d+k requests in 1s, .* MB read$/
```
Comment (thekemkid) [inline test/cli.test.js]: I know it _should_ be 1s, it _could_ be longer than 1s... could we change it to a selection group of [1|2]? any longer is a fail for sure :)

File: test/cli.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+const lines = [
+  /Running 1s test @ .*$/,
+  /10 connections.*$/,
+  /$/,
+  /Stat.*Avg.*Stdev.*Max.*$/,
+  /Latency \(ms\).*$/,
+  /Req\/Sec.*$/,
+  /Bytes\/Sec.*$/,
+  /$/,
+  /.* requests in 1s, .* read/
```
Comment (thekemkid) [inline test/cli.test.js]: I know that it _should_ be 1s, it _could_ be longer than 1s... could we change it to a selection group of [1|2]? any longer is a fail for sure :)

File: lib/progressTracker.js (hunk 1 above)
Comment (mcollina) [inline lib/progressTracker.js]: what we should probably do is to wrap those two blocks in functions, the main track function is hardly readable.

Comment (thekemkid) [inline lib/progressTracker.js]: Right, you can merge and I'll do that later :)

**Pull request merged**